porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,58 @@
+import tkinter
+import time
+from porcupine import tabs, get_tab_manager
+from porcupine.plugins import linenumbers
+
+# Dependent on code from linenumbers.py
+setup_after = ["linenumbers"]
+
+
+class AnchorManager:
+    def __init__(self, tab_textwidget: tkinter.Text, linenumber_instance: tkinter.Canvas) -> None:
+        self.tab_textwidget = tab_textwidget
+        self.linenumber_instance = linenumber_instance
+
+        self.anchor_symbol = "¶"
+
+        linenumber_instance.bind("<<Updated>>", self.update_linenumbers, add=True)
+
+    def toggle_on_off(self, event):
+        [cursor_index] = [self.tab_textwidget.index("insert linestart")]
+        for anchor in self.tab_textwidget.mark_names():
+            if anchor.startswith("anchor_") and self.tab_textwidget.index(anchor) == cursor_index:
```

Comment:
Anchor marks can be in the middle of the line. Suppose the text widget contains two lines, and there's an anchor mark in the middle of the second line. I'll denote an anchor mark with `|` even though it's actually invisible.

```
hello world
|blah blah
```

If I now go to end of first line and press delete, or beginning of second line and press backspace, it turns into this:

```
hello world|blah blah
```

There are a couple ways to handle this. Choose your favorite:
1. Before you do anything with anchor marks, loop through them all and move them to the beginning of the line if they aren't already. Something like `textwidget.mark_set(mark, f"{mark} linestart")`.
2. Change this check so that it checks whether the mark is on the same line. Something like this:
    ```
    if textwidget.compare("insert linestart", "<=", anchor) and textwidget.compare(anchor, "<=", "insert lineend"):
    ```
    Or like this:
    ```
    if textwidget.index("insert").split(".")[0] == textwidget.index(anchor).split(".")[0]:
    ```